lib/AsymmetricKey.js

```diff
@@ -105,6 +106,35 @@ export class AsymmetricKey {
     return JSON.parse(JSON.stringify(this._keyDescription));
   }
 
+  /**
+   * Gets an algorithm from a key description.
+   *
+   * @param {object} options - The options to use.
+   * @param {object} options.keyDescription - A key description.
+   *
+   * @returns {string} The resulting algorithm.
+   */
+  getAlgorithm({keyDescription}) {
+    const {publicKeyMultibase} = keyDescription;
+    // supported algorithms P-256, P-384, P-521, ed25519, secp256k1, P-256K
+    // @see https://github.com/multiformats/multicodec/blob/master/table.csv
+    // maps the bs58 prefix to the JOSE algorithm name
+    const algorithms = new Map([
+      ['6Mk', 'Ed25519'],
+      ['Dna', 'P-256'],
+      ['82L', 'P-384'],
+      ['2J9', 'P-521'],
+      // FIXME this prefix was not generated by an ed448 did key implementation
+      ['6GP', 'Ed448'],
+      ['Q3s', 'secp256k1']
```

Comment:
These should be prefixed with `z` to ensure there's no cross-encoding mix up.

```suggestion
      ['z6Mk', 'Ed25519'],
      ['zDna', 'P-256'],
      ['z82L', 'P-384'],
      ['z2J9', 'P-521'],
      // FIXME this prefix was not generated by an ed448 did key implementation
      ['z6GP', 'Ed448'],
      ['zQ3s', 'secp256k1']
```